client/src/components/App.jsx

```diff
@@ -109,11 +112,33 @@ class App extends React.Component {
     this.setState({
       favoriteProducts: updatedFavoriteProducts
     });
+  }
 
+  changeCurrentProduct(productId) {
+    let updatedProductInfo = this.cachedProducts[productId];
+    let {currentProduct, productStylesArray, reviews, questionList} = updatedProductInfo;
+    parserFunctions.getRelatedItems(productId, this.cachedProducts)
+      .then(relatedItems => {
+        relatedItems.filter(relatedItem => {
+          return !this.cachedProducts[relatedItem.currentProduct.id];
+        })
+          .forEach(relatedItem => {
+            this.cachedProducts[relatedItem.currentProduct.id] = relatedItem;
+          });
+        this.setState({
+          currentProductID: productId,
+          currentStyleID: productStylesArray[0].style_id,
+          currentProduct,
+          productStylesArray,
+          reviews,
+          questionList,
+          relatedItems
+        });
+      });
```

Comment:
super clean! Very cool to see this function working as well

Reply:
Thanks =)